qiskit_nature/operators/second_quantization/quadratic_hamiltonian.py

```diff
@@ -0,0 +1,306 @@
+            self._hermitian_part = np.zeros((self._n_orbitals, self._n_orbitals))
+        if self._antisymmetric_part is None:
+            self._antisymmetric_part = np.zeros((self._n_orbitals, self._n_orbitals))
+
+    @property
+    def hermitian_part(self) -> np.ndarray:
+        """The matrix of coefficients of terms that conserve particle number."""
+        return self._hermitian_part
+
+    @property
+    def antisymmetric_part(self) -> np.ndarray:
+        """The matrix of coefficients of terms that do not conserve particle number."""
+        return self._antisymmetric_part
+
+    @property
+    def constant(self) -> float:
+        """The constant."""
+        return self._constant
+
+    def _fermionic_op(self) -> FermionicOp:
+        """Convert to FermionicOp."""
+        terms = []
+        # TODO I shouldn't have to use string labels
+        # diagonal terms
+        for i in range(self._n_orbitals):
+            terms.append((f"+_{i} -_{i}", self.hermitian_part[i, i]))
+        # off-diagonal terms
+        for i in range(self._n_orbitals):
+            for j in range(i + 1, self._n_orbitals):
+                terms.append((f"+_{i} -_{j}", self.hermitian_part[i, j]))
+                terms.append((f"+_{j} -_{i}", self.hermitian_part[j, i]))
+                terms.append((f"+_{i} +_{j}", self.antisymmetric_part[i, j]))
+                terms.append((f"-_{j} -_{i}", self.antisymmetric_part[i, j].conj()))
+        # constant
+        terms.append(("", self.constant))
+        return FermionicOp(terms, register_length=self._n_orbitals)
+
+    def conserves_particle_number(self) -> bool:
+        """Whether the Hamiltonian conserves particle number."""
+        return np.allclose(self.antisymmetric_part, 0.0)
+
+    def majorana_form(self) -> Tuple[np.ndarray, float]:
+        r"""Return the Majorana representation of the Hamiltonian.
+
+        The Majorana representation of a quadratic Hamiltonian is
+
+        .. math::
+            \frac{i}{2} \sum_{j, k} A_{jk} f_j f_k + \text{constant}
+
+        where :math:`A` is a real antisymmetric matrix and the :math:`f_i` are
+        normalized Majorana fermion operators, which satisfy the relations:
+
+        .. math::
+            f_j = \frac{1}{\sqrt{2}} (a^\dagger_j + a_j)
+
+            f_{j + N} = \frac{i}{\sqrt{2}} (a^\dagger_j - a_j)
+
+        Returns:
+            - The matrix :math:`A`
+            - The constant
+        """
+        original = np.block(
+            [
+                [self.antisymmetric_part, self.hermitian_part],
+                [-self.hermitian_part.conj(), -self.antisymmetric_part.conj()],
+            ]
+        )
+        eye = np.eye(self._n_orbitals, dtype=complex)
+        majorana_basis = np.block([[eye, eye], [1j * eye, -1j * eye]]) / np.sqrt(2)
+        matrix = -1j * majorana_basis.conj() @ original @ majorana_basis.T.conj()
+        constant = 0.5 * np.trace(self.hermitian_part) + self.constant
+        # imaginary parts should be zero
```

Comment:
Do you want to assert this and raise an error if not?

Reply:
It's not necessary because it's guaranteed if the input matrices are valid, and if the user turned validation off, then this shouldn't be checked.